src/sap/fhir/model/r4/FHIRFilterOperatorUtils.js

```diff
@@ -228,9 +228,10 @@ sap.ui.define([
 		var sValue;
 		// special handling for string parameter as per fhir
 		// given eq "peter"
+		var sRegex = "[ \r\n\t\S]+";
 		if (sFilterValue && sFilterValue === FHIRFilterType.string) {
 			sValue = "\"" + vValue + "\"";
-		} else if (typeof vValue === "string" && isNaN(new Date(vValue).getTime())) {
+		} else if (typeof vValue === "string" && (vValue.match(sRegex) != null || isNaN(new Date(vValue).getTime()))) {
 			// incase of date as string it shouldnt be encoded
 			sValue = "\"" + vValue + "\"";
 		} else {
```

Comment:
could we rearrange this whole if else condition... we do both times the same... so condition could be merged

Reply:
we could also create a small private helper function contains only the datecheck
e.g 
````
containsDate : function (sValue) {
   return isNaN(new Date(vValue).getTime());
}
`````

Also I understood so far, that the regex should be chosen for all string values?

Reply:
@flovogt : converted into two helper functions which would determine if it needs to be encoded. the regex will test for all strings but match will occur only if the string has ws